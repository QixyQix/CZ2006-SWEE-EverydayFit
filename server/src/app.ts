import express from 'express';
import AuthRouter from './routes/authRouter';
import ForecastRouter from './routes/forecastRouter';
import FitnessPlanRouter from './routes/fitnessPlanRouter';
import ExerciseRouter from './routes/exerciseRouter';
import path from 'path';

const app = express();
app.use(express.json());

app.set('views', path.join(__dirname, 'views'));
app.set('view engine', 'ejs');

app.use(express.static(path.join(__dirname, 'public')));

// Configure routes
app.use('/auth', AuthRouter);
app.use('/forecasts', ForecastRouter);
app.use('/plan', FitnessPlanRouter);
app.use('/exercise', ExerciseRouter);

export {app as default};