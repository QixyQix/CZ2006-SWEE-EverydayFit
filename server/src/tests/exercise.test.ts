import testDB from "./mongoTestDB";
import app from '../app';
import request from "supertest";
import SeedDatabase from '../data/seed';
import ExerciseRepo from "../repo/exerciseRepo";

let exercises: any[] = [];
let createdExerciseID: string = "";

beforeAll(async () => {
    await testDB.connect();
    await SeedDatabase();

    await request(app).get('/exercise').then((response) => {
        exercises = response.body;
        for (const exercise of exercises) {
            if (exercise.name === 'Push Up') {
                createdExerciseID = exercise._id;
                break;
            }
        }
    })

});

afterEach(async () => {
    // await testDB.clearDatabase()
});

afterAll(async () => {
    await testDB.clearDatabase()
    await testDB.closeDatabase()
});

describe('EXERCISE: Get all exercises', () =>{
    it('Returns status 200 and all fitness details', async () => {
        await request(app).get('/exercise').expect((res) => {
            for (const exercise of res.body){
                if (exercise){
                    expect(exercise).toMatchSnapshot({
                        _id: expect.any(String),
                        name: expect.any(String),
                        category: expect.any(String) ,
                        outdoorOnly: expect.any(Boolean),
                        quantityType: expect.any(String),
                        quantityUnit: expect.any(String),
                        calorieBurnRatePerUnit: expect.any(Number)
                    })
                }
            }
        }).expect(200);
});

});

describe('EXERCISE: Get alternatives for exercise ID', () =>{
    it('Returns status 500 if non-existent exercise ID used', async() => {
        await request(app).get('/exercise/alt/12345abcde6789fghadc4f86').expect(500);
    });

    it('Returns null and status 200 if no exercise ID provided', async() => {
        await request(app).get('/exercise/alt/').expect(null).expect(200);
    });

    it('Returns status 200 and all fitness details', async () => {
        await request(app).get('/exercise').expect((res) => {
            for (const exercise of res.body){
                if (exercise){
                    expect(exercise).toMatchSnapshot({
                        _id: expect.any(String),
                        name: expect.any(String),
                        category: expect.any(String) ,
                        outdoorOnly: expect.any(Boolean),
                        quantityType: expect.any(String),
                        quantityUnit: expect.any(String),
                        calorieBurnRatePerUnit: expect.any(Number)
                    })
                }
            }
        }).expect(200);
    });
});

describe('EXERCISE: Get exercise by name', () =>{
    it('Returns null and status 200 if non-existent exercise name used', async() => {
        await request(app).get('/exercise/Fighting').expect(null).expect(200);
    });
    it ('Returns status 200 and details of exercise of exercise name', async() => {
        await request(app).get('/exercise/Push Up').expect((res) => {
            expect(res.body).toMatchSnapshot({
                _id: expect.any(String),
                name: expect.any(String),
                category: expect.any(String) ,
                outdoorOnly: expect.any(Boolean),
                quantityType: expect.any(String),
                quantityUnit: expect.any(String),
                calorieBurnRatePerUnit: expect.any(Number)
            })
        }).expect(200);
    })
});