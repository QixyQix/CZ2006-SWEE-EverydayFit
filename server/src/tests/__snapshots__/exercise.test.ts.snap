// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EXERCISE: Get all exercises Returns status 200 and all fitness details 1`] = `
Object {
  "__v": 0,
  "_id": Any<String>,
  "calorieBurnRatePerUnit": Any<Number>,
  "category": Any<String>,
  "name": Any<String>,
  "outdoorOnly": Any<Boolean>,
  "quantityType": Any<String>,
  "quantityUnit": Any<String>,
}
`;

exports[`EXERCISE: Get all exercises Returns status 200 and all fitness details 2`] = `
Object {
  "__v": 0,
  "_id": Any<String>,
  "calorieBurnRatePerUnit": Any<Number>,
  "category": Any<String>,
  "name": Any<String>,
  "outdoorOnly": Any<Boolean>,
  "quantityType": Any<String>,
  "quantityUnit": Any<String>,
}
`;

exports[`EXERCISE: Get all exercises Returns status 200 and all fitness details 3`] = `
Object {
  "__v": 0,
  "_id": Any<String>,
  "calorieBurnRatePerUnit": Any<Number>,
  "category": Any<String>,
  "name": Any<String>,
  "outdoorOnly": Any<Boolean>,
  "quantityType": Any<String>,
  "quantityUnit": Any<String>,
}
`;

exports[`EXERCISE: Get all exercises Returns status 200 and all fitness details 4`] = `
Object {
  "__v": 0,
  "_id": Any<String>,
  "calorieBurnRatePerUnit": Any<Number>,
  "category": Any<String>,
  "name": Any<String>,
  "outdoorOnly": Any<Boolean>,
  "quantityType": Any<String>,
  "quantityUnit": Any<String>,
}
`;

exports[`EXERCISE: Get all exercises Returns status 200 and all fitness details 5`] = `
Object {
  "__v": 0,
  "_id": Any<String>,
  "calorieBurnRatePerUnit": Any<Number>,
  "category": Any<String>,
  "name": Any<String>,
  "outdoorOnly": Any<Boolean>,
  "quantityType": Any<String>,
  "quantityUnit": Any<String>,
}
`;

exports[`EXERCISE: Get all exercises Returns status 200 and all fitness details 6`] = `
Object {
  "__v": 0,
  "_id": Any<String>,
  "calorieBurnRatePerUnit": Any<Number>,
  "category": Any<String>,
  "name": Any<String>,
  "outdoorOnly": Any<Boolean>,
  "quantityType": Any<String>,
  "quantityUnit": Any<String>,
}
`;

exports[`EXERCISE: Get all exercises Returns status 200 and all fitness details 7`] = `
Object {
  "__v": 0,
  "_id": Any<String>,
  "calorieBurnRatePerUnit": Any<Number>,
  "category": Any<String>,
  "name": Any<String>,
  "outdoorOnly": Any<Boolean>,
  "quantityType": Any<String>,
  "quantityUnit": Any<String>,
}
`;

exports[`EXERCISE: Get all exercises Returns status 200 and all fitness details 8`] = `
Object {
  "__v": 0,
  "_id": Any<String>,
  "calorieBurnRatePerUnit": Any<Number>,
  "category": Any<String>,
  "name": Any<String>,
  "outdoorOnly": Any<Boolean>,
  "quantityType": Any<String>,
  "quantityUnit": Any<String>,
}
`;

exports[`EXERCISE: Get alternatives for exercise ID Returns status 200 and all fitness details 1`] = `
Object {
  "__v": 0,
  "_id": Any<String>,
  "calorieBurnRatePerUnit": Any<Number>,
  "category": Any<String>,
  "name": Any<String>,
  "outdoorOnly": Any<Boolean>,
  "quantityType": Any<String>,
  "quantityUnit": Any<String>,
}
`;

exports[`EXERCISE: Get alternatives for exercise ID Returns status 200 and all fitness details 2`] = `
Object {
  "__v": 0,
  "_id": Any<String>,
  "calorieBurnRatePerUnit": Any<Number>,
  "category": Any<String>,
  "name": Any<String>,
  "outdoorOnly": Any<Boolean>,
  "quantityType": Any<String>,
  "quantityUnit": Any<String>,
}
`;

exports[`EXERCISE: Get alternatives for exercise ID Returns status 200 and all fitness details 3`] = `
Object {
  "__v": 0,
  "_id": Any<String>,
  "calorieBurnRatePerUnit": Any<Number>,
  "category": Any<String>,
  "name": Any<String>,
  "outdoorOnly": Any<Boolean>,
  "quantityType": Any<String>,
  "quantityUnit": Any<String>,
}
`;

exports[`EXERCISE: Get alternatives for exercise ID Returns status 200 and all fitness details 4`] = `
Object {
  "__v": 0,
  "_id": Any<String>,
  "calorieBurnRatePerUnit": Any<Number>,
  "category": Any<String>,
  "name": Any<String>,
  "outdoorOnly": Any<Boolean>,
  "quantityType": Any<String>,
  "quantityUnit": Any<String>,
}
`;

exports[`EXERCISE: Get alternatives for exercise ID Returns status 200 and all fitness details 5`] = `
Object {
  "__v": 0,
  "_id": Any<String>,
  "calorieBurnRatePerUnit": Any<Number>,
  "category": Any<String>,
  "name": Any<String>,
  "outdoorOnly": Any<Boolean>,
  "quantityType": Any<String>,
  "quantityUnit": Any<String>,
}
`;

exports[`EXERCISE: Get alternatives for exercise ID Returns status 200 and all fitness details 6`] = `
Object {
  "__v": 0,
  "_id": Any<String>,
  "calorieBurnRatePerUnit": Any<Number>,
  "category": Any<String>,
  "name": Any<String>,
  "outdoorOnly": Any<Boolean>,
  "quantityType": Any<String>,
  "quantityUnit": Any<String>,
}
`;

exports[`EXERCISE: Get alternatives for exercise ID Returns status 200 and all fitness details 7`] = `
Object {
  "__v": 0,
  "_id": Any<String>,
  "calorieBurnRatePerUnit": Any<Number>,
  "category": Any<String>,
  "name": Any<String>,
  "outdoorOnly": Any<Boolean>,
  "quantityType": Any<String>,
  "quantityUnit": Any<String>,
}
`;

exports[`EXERCISE: Get alternatives for exercise ID Returns status 200 and all fitness details 8`] = `
Object {
  "__v": 0,
  "_id": Any<String>,
  "calorieBurnRatePerUnit": Any<Number>,
  "category": Any<String>,
  "name": Any<String>,
  "outdoorOnly": Any<Boolean>,
  "quantityType": Any<String>,
  "quantityUnit": Any<String>,
}
`;

exports[`EXERCISE: Get exercise by name Returns status 200 and details of exercise of exercise name 1`] = `
Object {
  "__v": 0,
  "_id": Any<String>,
  "calorieBurnRatePerUnit": Any<Number>,
  "category": Any<String>,
  "name": Any<String>,
  "outdoorOnly": Any<Boolean>,
  "quantityType": Any<String>,
  "quantityUnit": Any<String>,
}
`;
